src/System.Management.Automation/engine/runtime/Operations/NumericOps.cs

```diff
@@ -863,5 +863,34 @@ internal static object CompareIne(char lhs, char rhs)
             char secondAsUpper = char.ToUpperInvariant(rhs);
             return firstAsUpper != secondAsUpper ? Boxed.True : Boxed.False;
         }
+
+        internal static object[] Range(char start, char end)
+        {
+            int lower = (int)start;
+            int upper = (int)end;
```

Comment:
I wonder - is this ASCII Arithmetic work correctly for unicode?

Reply:
It won't work with surrogate pairs because those can't be represented in a `char`, but I don't think that is an interesting scenario.

Reply:
Since we added `u{} should we detect surrogate pairs and write an error?

Reply:
I guess we should error on invalid characters (half of the pair). We could also support surrogate pair ranges if we didn't convert the string to char immediately, instead analyzing the string more thoroughly. But I'm not sure I'd block this PR on either of those.

Reply:
I guess it will work with traditional languages in common scenarios but before I saw issues from China and Japan mans - if the new operator does not matter to them I agree with you.

Reply:
@IISResetMe Could you please address the comment in code or add relevant comments in the PR description for future documentation?

Reply:
@iSazonov I don't feel confident enough in my unicode-fu to attempt to implement the appropriate handling in code for this at the moment, I've updated the PR description to reflect this deficiency instead.

Reply:
Thanks. 

Closed.